### PR TITLE
Hotfix for the DMan

### PR DESCRIPTION
**Pull request by marchermans**
This should hotfix the DMan not delivering Items properly.

I am currently on my Laptop my Tests were limited thanks to that, however interactions seem to work properly now.

### CODE REVIEW

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 83 issues
* ![MAJOR][MAJOR] 31 major
* ![MINOR][MINOR] 46 minor
* ![INFO][INFO] 6 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractBuilding.java#L38](https://github.com/Minecolonies/minecolonies/blob/d2d2d3c1e5661395f339f5efd81eac3cb2843c87/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L38): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 32 to the maximum authorized 20 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L38](https://github.com/Minecolonies/minecolonies/blob/d2d2d3c1e5661395f339f5efd81eac3cb2843c87/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L38): class "AbstractBuilding" has 68 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L161](https://github.com/Minecolonies/minecolonies/blob/d2d2d3c1e5661395f339f5efd81eac3cb2843c87/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L161): Remove this reference to "BuildingBaker". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L161](https://github.com/Minecolonies/minecolonies/blob/d2d2d3c1e5661395f339f5efd81eac3cb2843c87/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L161): Remove this reference to "BuildingBaker". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L162](https://github.com/Minecolonies/minecolonies/blob/d2d2d3c1e5661395f339f5efd81eac3cb2843c87/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L162): Remove this reference to "BuildingBlacksmith". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L162](https://github.com/Minecolonies/minecolonies/blob/d2d2d3c1e5661395f339f5efd81eac3cb2843c87/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L162): Remove this reference to "BuildingBlacksmith". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L163](https://github.com/Minecolonies/minecolonies/blob/d2d2d3c1e5661395f339f5efd81eac3cb2843c87/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L163): Remove this reference to "BuildingBuilder". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L164](https://github.com/Minecolonies/minecolonies/blob/d2d2d3c1e5661395f339f5efd81eac3cb2843c87/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L164): Remove this reference to "BuildingHome". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L164](https://github.com/Minecolonies/minecolonies/blob/d2d2d3c1e5661395f339f5efd81eac3cb2843c87/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L164): Remove this reference to "BuildingHome". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L165](https://github.com/Minecolonies/minecolonies/blob/d2d2d3c1e5661395f339f5efd81eac3cb2843c87/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L165): Remove this reference to "BuildingFarmer". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit: 
Will we need that for 1.11 as well?

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/940/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java [hunk 1]
```diff
@@ -301,33 +300,26 @@ private AIState deliver()
                     return DELIVERY;
                 }
 
-                final InventoryCitizen workerInventory = worker.getInventoryCitizen();
+                final InvWrapper workerInventory = new InvWrapper(worker.getInventoryCitizen());
                 for (int i = 0; i < new InvWrapper(worker.getInventoryCitizen()).getSlots(); i++)
                 {
-                    final ItemStack stack = workerInventory.getStackInSlot(i);
-                    if (stack == null)
+                    final ItemStack stack = workerInventory.extractItem(1, Integer.MAX_VALUE, false);
```
Comment (xavierh) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java]: extractItem(**1**, Integer.MAX_VALUE, false);

Is this correct ?

Comment (marchermans) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java]: Ah no you are right that 1 should be an i. Will fix asap.

**Comment by marchermans** [comment]
@Raycoms @Kostronor @xavierh Done
This should be good to go.

**Comment by marchermans** [comment]
@Raycoms yes you will Need this for 1.11 as well. The DMan is just not getting enough Information to do his Job properly.

On top this fixes some behaviour issues in the Insertion methods.

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 83 issues
* ![MAJOR][MAJOR] 31 major
* ![MINOR][MINOR] 46 minor
* ![INFO][INFO] 6 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractBuilding.java#L38](https://github.com/Minecolonies/minecolonies/blob/ec69b333903482786e2b919981b13f336da7d47b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L38): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 32 to the maximum authorized 20 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L38](https://github.com/Minecolonies/minecolonies/blob/ec69b333903482786e2b919981b13f336da7d47b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L38): class "AbstractBuilding" has 68 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L161](https://github.com/Minecolonies/minecolonies/blob/ec69b333903482786e2b919981b13f336da7d47b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L161): Remove this reference to "BuildingBaker". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L161](https://github.com/Minecolonies/minecolonies/blob/ec69b333903482786e2b919981b13f336da7d47b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L161): Remove this reference to "BuildingBaker". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L162](https://github.com/Minecolonies/minecolonies/blob/ec69b333903482786e2b919981b13f336da7d47b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L162): Remove this reference to "BuildingBlacksmith". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L162](https://github.com/Minecolonies/minecolonies/blob/ec69b333903482786e2b919981b13f336da7d47b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L162): Remove this reference to "BuildingBlacksmith". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L163](https://github.com/Minecolonies/minecolonies/blob/ec69b333903482786e2b919981b13f336da7d47b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L163): Remove this reference to "BuildingBuilder". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L164](https://github.com/Minecolonies/minecolonies/blob/ec69b333903482786e2b919981b13f336da7d47b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L164): Remove this reference to "BuildingHome". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L164](https://github.com/Minecolonies/minecolonies/blob/ec69b333903482786e2b919981b13f336da7d47b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L164): Remove this reference to "BuildingHome". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
1. ![MAJOR][MAJOR] [AbstractBuilding.java#L165](https://github.com/Minecolonies/minecolonies/blob/ec69b333903482786e2b919981b13f336da7d47b/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L165): Remove this reference to "BuildingFarmer". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/940/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by marvin-bitterlich** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/940/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**